src/GivingCircle.Api.DataAccess/Repositories/UserRepository.cs

```diff
@@ -0,0 +1,128 @@
+﻿using Dapper;
+using GivingCircle.Api.DataAccess.Client;
+using GivingCircle.Api.DataAccess.Responses;
+using GivingCircle.Api.Models;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GivingCircle.Api.DataAccess.Repositories
+{
+    public class UserRepository : IUserRepository
+    {
+        private readonly PostgresClient _postgresClient;
+
+        /// <summary>
+        /// Initializes an instance of the <see cref="UserRepository"/> class
+        /// </summary>
+        /// <param name="postgresClient">The postgres client</param>
+        public UserRepository(PostgresClient postgresClient)
+        {
+            _postgresClient = postgresClient;
+        }
+
+        //USER METHODS
+
+        public async Task<GetUserResponse> GetUserAsync(string userId)
+        {
+            // Object to map the parameters to the query
+            object parameters = new { User_Id = userId };
+
+            var user = await _postgresClient.QuerySingleAsync<GetUserResponse>("SELECT * FROM users WHERE user_id = @User_Id", parameters);
+
+            return user ?? null;
+        }
+
+        public async Task<bool> CreateUserAsync(User user)
+        {
+            // The string builder
+            StringBuilder queryBuilder = new();
+
+            // This represents the number of rows effected by our query
+            int createdResult;
+
+            // Construct the query
+            var query = queryBuilder
+                .Append($"INSERT INTO users")
+                .Append("(user_id, first_name, middle_initial, last_name, password, email, created_date, planned_end_date)")
+                .Append("VALUES (@UserId, @FirstName, @MiddleInitial, @LastName, @Password, @email)")
+                .ToString();
+
+            createdResult = await _postgresClient.ExecuteAsync(query, user);
+
+            // If we created 1 new user then we succeeded
+            return (createdResult == 1);
+        }
+
+        public async Task<bool> UpdateUserAsync(string userId, User user)
+        {
+            // The string builder
+            StringBuilder queryBuilder = new();
+
+            // This represents the number of rows effected by our query
+            int updatedResult;
+
+            // The dynamic parameters bag
+            Dictionary<string, object> parametersDictionary = new()
+            {
+                { "@FirstName",  user.FirstName},
+                { "@MiddleInitial",  user.MiddleInitial},
+                { "@LastName", user.LastName},
+                { "@Password", user.Password },
+                { "@Email", user.Email },
+                { "@UserId", userId }
```

Comment:
The user won't be able to change their user id once it's generated by the controller

Reply:
This is using userID to identify user for updates.

Reply:
Ah my bad I didn't read the query close enough